### PR TITLE
Improve documentation of `dask.config.set`

### PR DESCRIPTION
**Pull request by crusaderky**
Closes https://github.com/dask/distributed/issues/5752

### CODE REVIEW

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -168,18 +168,23 @@ and interprets ``"."`` as nested access:
...
 
-   with dask.config.set({'scheduler.work-stealing': True}):
-       ...
+   >>> with dask.config.set({'optimization.fuse.ave-width': 4}):
+   ...     arr2, = dask.optimize(arr)
 
 Note that the ``set`` function treats underscores and hyphens identically.
-For example, ``dask.config.set({'scheduler.work-stealing': True})`` is
-equivalent to ``dask.config.set({'scheduler.work_stealing': True})``.
+For example, ``dask.config.set({'optimization.fuse.ave_width': 4)`` is
```
Comment (bryanwweber) [inline docs/source/configuration.rst:181]:
```suggestion
For example, ``dask.config.set({'optimization.fuse.ave_width': 4})`` is
```

File: docs/source/configuration.rst [hunk 2]
```diff
@@ -168,18 +168,23 @@ and interprets ``"."`` as nested access:
...
+equivalent to ``dask.config.set({'optimization.fuse.ave-width': 4})``.
+
+Finally, note that persistent objects may acquire configuration settings at init
+time and cache them for performance reasons. This is particularly true for
+``dask.distributed`` objects (Client, Scheduler, Worker, Nanny, etc.)
```
Comment (bryanwweber) [inline docs/source/configuration.rst:186]: One suggestion here and a comment. The comment is in the last line, can we avoid using `etc.`? For a new user, they do not know what can be contained within that `etc`. I know we don't want to list every possible object, but this list already seems like a fairly complete list that users would interact with regularly. Perhaps something like `dask.distributed objects such as Clients, Schedulers, Workers, and Nannys.`?

```suggestion
Finally, note that persistent objects may acquire configuration settings when
they are initialized. These settings may also be cached for performance reasons.
This is particularly true for ``dask.distributed`` objects (Client, Scheduler, Worker, Nanny, etc.)
```

**Review by bryanwweber — COMMENTED** [review]
Thanks @crusaderky! Just a couple small comments here

File: docs/source/configuration.rst [hunk 3]
```diff
@@ -168,18 +168,23 @@ and interprets ``"."`` as nested access:
...
 
 Note that the ``set`` function treats underscores and hyphens identically.
-For example, ``dask.config.set({'scheduler.work-stealing': True})`` is
-equivalent to ``dask.config.set({'scheduler.work_stealing': True})``.
+For example, ``dask.config.set({'optimization.fuse.ave_width': 4})`` is
+equivalent to ``dask.config.set({'optimization.fuse.ave-width': 4})``.
+
+Finally, note that persistent objects may acquire configuration settings when
+they are initialized. These settings may also be cached for performance reasons.
+This is particularly true for ``dask.distributed`` objects (Client, Scheduler, Worker, Nanny, etc.)
```
Comment (crusaderky) [inline docs/source/configuration.rst:186]:
```suggestion
This is particularly true for ``dask.distributed`` objects such as Client, Scheduler,
Worker, and Nanny.
```

**Comment by bryanwweber** [comment]
Thanks @crusaderky!

**Pull request merged**